include/engine/guidance/assemble_leg.hpp

```diff
@@ -54,7 +62,8 @@ std::array<std::uint32_t, SegmentNumber> summarizeRoute(const std::vector<PathDa
                 *out = *in;
             }
         }
-        return out;
+        BOOST_ASSERT(out != end);
+        return ++out;
```

Comment:
close. It accumulates duration and does something else. the something else seems like it should be unique, but it does not work. I don't know exactly what it is supposed to be.